resources/storage.proto

```diff
@@ -11,7 +34,42 @@ enum Action {
 	DELETE = 4;
 }
 
-// Storage Query from client to cluster
+/*
+ * ============Metadata============ 
+ * This class is for storing metadata. We
+ * should store sequence size so that we can successfully reconstruct
+ * complete file/data from chunks. (Basically, we should know when we should
+ * write chunks to the file and so we should know what is the sequence
+ * size).
+ *
+ * So, for example, if a file has 8 chunks of data. seq_size should be 8 +
+ * 1. (1 for metadata chunk).
+ *
+ * size attribute is for storing file size in bytes. (optional) time will
+ * store creation time. (optional)
+ *
+ * Metadata chunk should be created by client program. To create it, client
+ * should first identify file size and how many chunks will be created from
+ * the file.
+ *
+ * Metadata should be stored as sequenceNo 0.
+ */
+message Metadata {
+	required int32 seq_size = 19;
+	optional int32 size = 20;
+	optional int64 time = 21;
+}
+
+
+/*
+ * ============Query============
+ * 
+ * Query is command for the cluster. It should contain 
```

Comment:
Query can be used for internal communication. Workmessage and worktask should carry this query. But still it is a command (action) for the cluster.